esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -52,63 +52,145 @@ LogAccessTimeRange requestedRangeToLARange(IConstTimeRange & reqrange)
+{
+    if (isEmptyString(searchByValue) && searchByCategory != CLogAccessType_All)
+       throw makeStringException(-1, "WsLogAccess: Empty searchByValue detected");
 
-    LogAccessConditions logFetchOptions;
     switch (searchByCategory)
     {
         case CLogAccessType_All:
-            logFetchOptions.setFilter(getWildCardLogAccessFilter());
-            break;
+            return getWildCardLogAccessFilter();
         case CLogAccessType_ByJobIdID:
-            logFetchOptions.setFilter(getJobIDLogAccessFilter(searchByValue));
-            break;
+            return getJobIDLogAccessFilter(searchByValue);
         case CLogAccessType_ByComponent:
-            logFetchOptions.setFilter(getComponentLogAccessFilter(searchByValue));
-            break;
+            return getComponentLogAccessFilter(searchByValue);
         case CLogAccessType_ByLogType:
         {
             LogMsgClass logType = LogMsgClassFromAbbrev(searchByValue);
             if (logType == MSGCLS_unknown)
                 throw makeStringExceptionV(-1, "Invalid Log Type 3-letter code encountered: '%s' - Available values: 'DIS,ERR,WRN,INF,PRO,MET'", searchByValue);
 
-            logFetchOptions.setFilter(getClassLogAccessFilter(logType));
-            break;
+            return getClassLogAccessFilter(logType);
         }
         case CLogAccessType_ByTargetAudience:
         {
             MessageAudience targetAud = LogMsgAudFromAbbrev(searchByValue);
             if (targetAud == MSGAUD_unknown || targetAud == MSGAUD_all)
                 throw makeStringExceptionV(-1, "Invalid Target Audience 3-letter code encountered: '%s' - Available values: 'OPR,USR,PRO,ADT'", searchByValue);
 
-            logFetchOptions.setFilter(getAudienceLogAccessFilter(targetAud));
-            break;
+            return getAudienceLogAccessFilter(targetAud);
         }
         case CLogAccessType_BySourceInstance:
         {
-            logFetchOptions.setFilter(getInstanceLogAccessFilter(searchByValue));
-            break;
+            return getInstanceLogAccessFilter(searchByValue);
         }
         case CLogAccessType_BySourceNode:
         {
-            logFetchOptions.setFilter(getHostLogAccessFilter(searchByValue));
+            return getHostLogAccessFilter(searchByValue);
             break;
         }
+        case CLogAccessType_ByFieldName:
+        {
+            return getColumnLogAccessFilter(serchField, searchByValue);
+        }
         case LogAccessType_Undefined:
         default:
             throw makeStringException(-1, "Invalid remote log access request type");
     }
+}
+
+ILogAccessFilter * processLogFilter(IConstLogFilter * logFilter)
+{
+    if (logFilter)
+        return processLogFilterByFields(logFilter->getLogCategory(), logFilter->getSearchByValue(), logFilter->getSearchField());
+    else
+        return nullptr;
+}
+
+ILogAccessFilter * processBinaryLogFilter(IConstBinaryLogFilter * binaryfilter)
```

Comment:
You could argue that "processing" happens inside 

`m_remoteLogAccessor->fetchLog(logFetchOptions, logcontent, logAccessFormatFromName(req.getFormat()));
`
And that the name of this function would be 
buildLogAccessFilter(IConstBinaryLogFilter * binaryfilter);
and above
`buildLogAccessFilter(IConstLogFilter * logFilter);`
Named after what they were building rather than the input.

The main point being that it becomes intuitive that this is the preparation stage, and processing happens further in.